task20/js/main.js

```diff
@@ -0,0 +1,88 @@
+import '../css/style.css'
+import { createElementWithClassName } from './utils/createElement'
+import { renderLoading } from './modules/loading'
+import { removeLoading } from './modules/loading'
+import { renderErrorMessage } from './modules/error'
+
+const endpoint = "https://mocki.io/v1/7afded4f-46ed-4535-b770-997abc352aea";
+
+const parent = document.getElementById("js-parent");
+
+const getFetchData = async () => {
+    const response = await fetch(endpoint);
+    if (!response.ok){
+        console.error(`$(response:status):$(response:statusText)`);
+        renderErrorMessage(parent, "問題が発生し表示することができません。");
+    }
+    const json = await response.json();
+    return json.data;
+};
+
+const fetchTableData = () => {
+    return new Promise((resolve) => {
+        setTimeout(() => {
+            resolve(getFetchData(endpoint));
+        }, 3000);
+    });
+};
+
+const initialize = async () => {
+    renderLoading(parent);
+    let userContentsData
+    try {
+        userContentsData = await fetchTableData();
+    } catch(error) {
+        renderErrorMessage(parent, "問題が発生し表示することができません。");
+    } finally {
+        removeLoading(parent);
+    }
+    renderTableContents(userContents, userContentsData);
+};
+initialize();
+
+const userContents =  {
+    "userId" : "ID",
+    "userName" : "名前",
+    "userGender" : "性別",
+    "userAge" : "年齢"
+};
```

Comment:
IMO
If the object name is telling you something, I don't think you need to repeat that in the property name.
```suggestion
const userContents =  {
    "id" : "ID",
    "name" : "名前",
    "gender" : "性別",
    "age" : "年齢"
};
```

Reply:
Thank you for pointing this out. You are correct. I have corrected it.
[d2ed482](https://github.com/seeeen044/JavaScript_handson/pull/21/commits/d2ed48226a741550b2f99062192726cd7079df78)